# ParaMut

## Release files description:

### Supplementary Material.2  

A table containing all paralogous domains obtained through blasp. Description of columns:  src_chr (chromosome where the source gene is located), src_pos (genomic position where the source variant is located), src_gene (source gene name), src_tx_id (transcript id of the source gene), src_aapos (amino acid position in the transcript of the source gene) , src_AminoAcid (the aminoacid present in src_Aminoacid of the source gene), src_start_whole_domain (the start position of the source domain), src_end_whole_domain (the end position of the source domain) , src_protein_sequence (the aminoacid sequence of the source domain), para_chr (chromosome where the paralogous gene is located, para_pos (genomic position where the paralogous variant is located), para_gene(paralogous gene name) , para_tx_id (transcript id of the paralogous gene), para_aapos (amino acid position in the transcript of the paralogous gene), para_start_whole_domain (the start position of the paralogous domain), para_end_whole_domain (the end position of the paralogous domain), para_protein_sequence (the aminoacid sequence of the paralogous domain).

### Supplementary Material.3 

A table containing results from merging Supplementary material 2 with ClinVar database through source genes. Description of columns: src_chr (chromosome where the source gene is located), src_pos (genomic position where the source variant is located) , src_ref (reference nucleic acid of the source variant), src_alt (alternative nucleic acid in the source variant) , src_gene_name (source gene name), src_tx_id (transcript id of the source gene), src_aapos (amino acid position in the transcript of the source gene) , src_AminoAcid (the mutated aminoacid of the source gene), src_start_whole_domain (the start position of the source domain), src_end_whole_domain (the end position of the source domain) , src_protein_sequence (the aminoacid sequence of the source domain), src_disease_cln (the disease caused by the variant in the source gene) , src_CLINSIG (the status of source variant either “Pathogenic” or “Likely_pathogenic”), para_chr (chromosome where the paralogous gene is located, para_pos (genomic position where the paralogous variant is located), para_gene_name (paralogous gene name) , para_tx_id (transcript id of the paralogous gene), para_aapos (amino acid position in the transcript of the paralogous gene), para_start_whole_domain (the start position of the paralogous domain), para_end_whole_domain (the end position of the paralogous domain), para_protein_sequence (the aminoacid sequence of the paralogous domain). 

### Revel_t.test_database.csv  

A table used for preforming a two sided t.test on REVEL scores. Description of columns: src_chr (chromosome where the source gene is located), src_pos (genomic position where the source variant is located), src_ref (reference nucleic acid of the source variant), src_alt (alternative nucleic acid in the source variant), src_gene_name (source gene name), src_ClinVar (the status of source variant either “Pathogenic”, “Likely_pathogenic”, "Benign", or "Likely_benign"), para_chr (chromosome where the paralogous gene is located, para_pos (genomic position where the paralogous variant is located), para_ref (reference nucleic acid of the paralogue variant), para_alt (alternative nucleic acid in the paralogue variant), para_gene_name (paralogous gene name), para_ClinVar (the status of paralogue variant either “Pathogenic”, “Likely_pathogenic”, "Benign", or "Likely_benign"), para_REVEL (REVEL score of the paralogue variant).

### Chi_square_pathogenic.csv 

A table used for preforming a chi square test. Description of columns: src_chr (chromosome where the source gene is located), src_pos (genomic position where the source variant is located), src_ref (reference nucleic acid of the source variant), src_alt (alternative nucleic acid in the source variant), src_gene_name (source gene name), src_ClinVar (the status of source variant either “Pathogenic” or “Likely_pathogenic”), para_chr (chromosome where the paralogous gene is located, para_pos (genomic position where the paralogous variant is located), para_ref (reference nucleic acid of the paralogue variant), para_alt (alternative nucleic acid in the paralogue variant), para_gene_name (paralogous gene name), para_ClinVar (the status of paralogue variant either “Pathogenic”, “Likely_pathogenic”, "Benign", or "Likely_benign"), src_REVEL (REVEL score of the source variant), para_REVEL (REVEL score of the paralogue variant).

### Chi_square_benign.csv 

A table used for preforming a chi square test. Description of columns: src_chr (chromosome where the source gene is located), src_pos (genomic position where the source variant is located), src_ref (reference nucleic acid of the source variant), src_alt (alternative nucleic acid in the source variant), src_gene_name (source gene name), src_ClinVar (the status of source variant either “Benign” or “Likely_benign”), para_chr (chromosome where the paralogous gene is located, para_pos (genomic position where the paralogous variant is located), para_ref (reference nucleic acid of the paralogue variant), para_alt (alternative nucleic acid in the paralogue variant), para_gene_name (paralogous gene name), para_ClinVar (the status of paralogue variant either “Pathogenic”, “Likely_pathogenic”, "Benign", or "Likely_benign"), src_REVEL (REVEL score of the source variant), para_REVEL (REVEL score of the paralogue variant).
